flink-libraries/flink-table/src/main/scala/org/apache/flink/table/functions/utils/UserDefinedFunctionUtils.scala

```diff
@@ -329,6 +337,41 @@ object UserDefinedFunctionUtils {
     }
   }
 
+
+  /**
+    * Internal method of AggregateFunction#getAccumulatorType() that does some pre-checking
+    * and uses [[TypeExtractor]] as default return type inference.
+    */
+  def getAccumulatorTypeOfAggregateFunction(
+    aggregateFunction: AggregateFunction[_, _],
```

Comment:
I find that only the following code block can be extracted. I think maybe we don't need to extract it. What do you think? @sunjincheng121 

```
  val resultType = try {
      val method: Method = aggregateFunction.getClass.getMethod(methodName)
      method.invoke(aggregateFunction).asInstanceOf[TypeInformation[_]]
    } catch {
      case _: NoSuchMethodException => null
      case ite: Throwable => throw new TableException("Unexpected exception:", ite)
    }
```